### PR TITLE
dev: Run policy tests via `just`

### PR DESCRIPTION
**Pull request by olix0r**
The `policy-test/README.md` file includes instructions for manually
running policy-controller tests. This change replaces these instructions
with a `justfile` that can provision a test cluster in k3d, load test
dependencies and execute tests via `just policy-test`.

The policy controller CI workflow is *not* yet updated to use this same
tooling, but that should be done in a followup. The rust workflow,
however, uses the `justfile` to run basic tests.

This change also renames the *Rust Static Checks* workflow to simply
*Rust*. The new name is less verbose and more accurate: the workflow
includes unit tests, which aren't really 'static'.

### CODE REVIEW

File: justfile [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+export DOCKER_REGISTRY := env_var_or_default("DOCKER_REGISTRY", "test-" + _test-id + ".local/linkerd")
+
+# The docker image tag.
+image-tag := `bin/root-tag`
+
+# The Linkerd CLI binary.
+linkerd := "bin/linkerd"
+
+_ctx := "--context=k3d-" + test-cluster-name
+_linkerd :=  linkerd + " " + _ctx
```
Comment (mateiidavid) [inline justfile:16]: Little indent issue here

```suggestion
_linkerd := linkerd + " " + _ctx
```

File: justfile [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+        serverauthorizations.policy.linkerd.io \
+        servers.policy.linkerd.io
+
+# Build/fetch the Linkerd containers and load them onto the test cluster.
+_policy-test-images: docker-pull-policy-test-deps docker-build-policy-controller && policy-test-load-images
+    bin/docker-build-controller
+    bin/docker-build-proxy
+
+docker-pull-policy-test-deps:
+    docker pull -q docker.io/bitnami/kubectl:latest
```
Comment (mateiidavid) [inline justfile:167]: Probably not important in the grand scheme of things but to make sure client & server versions match would it make sense to use `test-cluster-k8s` here?, e.g

`docker.io/bitnami/kubectl:{{test-cluster-k8s}}`

Or would it be better not to be so tightly coupled here?

File: .github/workflows/rust.yml [hunk 3]
```diff
@@ -27,9 +28,10 @@ jobs:
     container:
       image: docker://rust:1.62.0
     steps:
+      - uses: extractions/setup-just@aa5d15c144db4585980a44ebfdd2cf337c4f14cb
```
Comment (mateiidavid) [inline .github/workflows/rust.yml:31]: Would it make sense to specify the just version here to make sure it's the same one we use in the devcontainer?

File: justfile (hunk 2 above)
Comment (olix0r) [reply justfile:167]: this is pre-pulling images referenced in the tests. So, we'd have to somehow configure that version in the tests, too... https://github.com/linkerd/linkerd2/blob/5d0ffa71b395c12fb4a23693fd0b80c4da17aecc/policy-test/src/curl.rs#L161

i think it's fine to just use `latest` unless/until we hit compatibility issues.

File: .github/workflows/rust.yml (hunk 3 above)
Comment (mateiidavid) [reply .github/workflows/rust.yml:31]: Makes sense!

File: justfile [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+            --set 'policyController.logLevel=info\,linkerd=trace\,kubert=trace' \
+        | {{ _kubectl }} apply -f -
+    {{ _linkerd }} check -o short --wait=1m
+
+# Install CRDs on the test cluster.
+test-cluster-install-crds: _test-cluster-exists && _test-cluster-crds-ready
+    {{ _linkerd }} install --crds | {{ _kubectl }} apply -f -
+
+_test-cluster-crds-ready:
+    {{ _kubectl }} wait --for condition=established --timeout=60s crd \
```
Comment (adleong) [inline justfile:154]: could use `linkerd check --crds` if you prefer

**Review by adleong — APPROVED** [review]

**Pull request merged**